sklearn/sklearn-template/template/README.md

```diff
@@ -0,0 +1,105 @@
+        |__ utils.py            # utility functions including e.g. loading data from bigquery and cloud storage
+        |__ task.py             # training job entry point, handling the parameters passed from command line 
+    |__ hptuning_config.yaml    # config file for running hyperparameter tunning 
+    |__ setup.py                # specify necessary dependency of the training job
+    |__ requirements.txt        # specify necessary dependency of the training job
+    |__ Dockerfile
+```
+
+# Steps to make use of the template
+## Step 0. Prerequisite
+Before you follow the instructions below to adapt the tempate to your machine learning job, 
+you need a Google cloud project if you don't have one. You can find detailed instructions 
+[here](https://cloud.google.com/dataproc/docs/guides/setup-project).
+
+- Make sure the following API & Services are enabled.
+    * Cloud Storage
+    * Cloud Machine Learning Engine
+    * BigQuery API
+    * Cloud Build API (for CI/CD integration)
+    * Cloud Source Repositories API (for CI/CD integration)
+
+- Configure project id as environment variable
+  ```bash
+  $ export PROJECT_ID=[your-google-project-id]
+  ```
+  
+## Step 1. Modify metadata.py
+```python
+# Example for iris dataset
+COLUMNS = None  # Schema of the data. Necessary for data stored in GCS
+
+NUMERIC_FEATURES = [
+    'sepal_length',
+    'sepal_width',
+    'petal_length',
+    'petal_width',
+]
+
+CATEGORICAL_FEATURES = [
+
+]
+
+FEATURE_COLUMNS = NUMERIC_FEATURES + CATEGORICAL_FEATURES
+
+LABEL = 'species'
+
+METRIC_FILE_NAME_PREFIX = 'metric'
+MODEL_FILE_NAME_PREFIX = 'model'
+MODEL_FILE_NAME_SUFFIX = '.joblib'
+
+BASE_QUERY = '''
+    Select * From `{table}`
+  '''
+```
+
+metadata.py is where the dataset's metadata is defined. The code snippets above is an example configured 
+for iris dataset (can be found at `bigquery-public-data.ml_datasets.iris`). In most cases, only the following
+items need to be modified, in order to adapt to the target dataset. 
+- **COLUMNS**: the schema of ths data, only required for data stored in GCS
+- **NUMERIC_FEATURES**: columns those will be treated as numerical features
+- **CATEGORICAL_FEATURES**: columns those will be treated as categorical features
+- **LABEL**: column that will be treated as label
+
+## Step 2. Submit ML training job
+```shell
+bash scripts/train.sh [INPUT_PATH] [RUN_ENV] [BUCKET_NAME]
+```
+- INPUT_PATH: Dataset to use for training and evaluation. Can be BigQuery table or a file (CSV).
+              BigQuery table should be specified as `PROJECT_ID.DATASET.TABLE_NAME`.
+- RUN_ENV: Whether to run `local` (on-prem) or `remote` (GCP).
+- BUCKET_NAME: GSC bucket to be used for storing the trained model and evaluation result.
+
+## Step 3. Deploy the trained scikit-learn model
+After training finishes, the model will be exported to specified job directory in Google Cloud Storage. 
```

Comment:
Nonblock:
I was thinking of adding a script for deploying the model. With that, we can potentially do the CD part of CI/CD.